docs/optimization/improve-parallelism/dataflow.md

```diff
@@ -1,8 +1,8 @@
 ### Enabling dataflow parallelism
 
-This guide teaches what data parallelism is and how it can improve the execution time of Concrete circuits.
+This guide explains data parallelism and how it can improve the execution time of **Concrete** circuits.
 
-Dataflow parallelism is a great feature, especially when the circuit is doing a lot of scalar operations.
+Dataflow parallelism is particularly useful when the circuit performs a lot of scalar operations. 
```

Comment:
that is no actually related to scalar or tensor operators. A better explanation could.
`Dataflow parallelism is particularly useful when the circuit contains tasks which are not interdependent`, maybe @antoniupop you may have a better formulation.

Reply:
Indeed - I would stress the fact that dataflow is particularly interesting when we don't have loop/doall parallelism. E.g.: "Dataflow parallelism is particularly useful when the circuit performs computations that are neither completely independent (such as loop/doall parallelism) nor fully dependent (e.g. sequential, non-parallelizable code). In such cases dataflow tasks can execute as soon as their inputs are available and thus minimizing over-synchronization."

Reply:
added thx